### PR TITLE
HIP: Long Term Scheduled Transactions

### PR DESCRIPTION
**Pull request by statop**
**Description**:

Adds a HIP for the introduction of long term scheduled transactions

**Related issue(s)**:

Fixes #

**Notes for reviewer**:=

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by rbair23 — APPROVED** [review]

File: HIP/hip-0000-long-term-scheduled-transactions.md [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+  - `ScheduleCreate` shall remain at a $0.01 fee for all other transaction types.
+  - **Fees will still be dynamic based on resource usage.**
+  
+### Transaction Types
+
+- The `scheduling.whitelist` setting shall be updated to - `ConsensusSubmitMessage,CryptoTransfer,TokenMint,TokenBurn,CryptoCreate,CryptoUpdate,FileUpdate,SystemDelete,SystemUndelete,Freeze,ContractCall,ContractCreate,ContractUpdate,ContractDelete`
+
+### Mirror Node Change
+
+- Add `waitForExpiry` to the mirror node schedule api.
```
Comment (steven-sheehy) [inline HIP/hip-0000-long-term-scheduled-transactions.md:348]: We should also add `expirationTime` to the API.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: HIP/hip-0000-long-term-scheduled-transactions.md (hunk 1 above)
Comment (statop) [reply HIP/hip-0000-long-term-scheduled-transactions.md:348]: It’s already there.

Comment (steven-sheehy) [reply HIP/hip-0000-long-term-scheduled-transactions.md:348]: This is the JSON output of `/api/v1/schedules/{id}`. I don't see `expiration_time`, which is a new field you added in this HIP:

```json
{
   "admin_key":null,
   "deleted":false,
   "consensus_timestamp":"1620327749.254652999",
   "creator_account_id":"0.0.950",
   "executed_timestamp":"1620327749.254653000",
   "memo":"",
   "payer_account_id":"0.0.950",
   "schedule_id":"0.0.226412",
   "signatures":[
      {
         "consensus_timestamp":"1620327749.254652999",
         "public_key_prefix":"Qg==",
         "signature":"SUZHg7uixmb5MjubWNSs4H1FRQdgHwgj9KEe0XWQD4yvD9W+AAQkZBy+FdOnYbz/bfTT3op++Opg\nfGKKRqYKAg==",
         "type":"ED25519"
      }
   ],
   "transaction_body":"CIDC1y8SFFNob2NrZWQsIEkgdGVsbCB5b3UhShMKEQoGCgIYYhACCgcKAxi2BxAB"
}
```

Comment (statop) [reply HIP/hip-0000-long-term-scheduled-transactions.md:348]: You are correct, sorry. Will add. I have been staring at the services code too much.

**Review by steven-sheehy — APPROVED** [review]
LGTM

File: HIP/hip-0000-long-term-scheduled-transactions.md [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+
+- We cannot guarantee Scheduled Transactions will execute exactly at their scheduled "consensus time".
+  - There is no way to prevent another transaction from taking a given nanosecond slot without massive changes to the entire system.
+  - Things like throttling and other system factors also interfere with when we can actually execute a transaction.
+
+_Scheduled Transactions will execute at the earliest available consensus time after their `expirationTime` on a best-effort basis._
+
+### Throttling
+
+- There are throttles on how many transactions can execute per round.
```
Comment (tinker-michaelj) [inline HIP/hip-0000-long-term-scheduled-transactions.md:64]: Strictly speaking, throttles limit either operations-per-second (where "operation" means either "query" or "transaction"); or EVM gas-per-second.

File: HIP/hip-0000-long-term-scheduled-transactions.md [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+
+_The transaction ID of the child can be found deterministically at `ScheduleCreate` time_
+
+_Users can also use the `executed_timestamp` in the mirror node schedule API to find child transactions via the mirror node transactions API._
+
+### Corner cases
+
+- An account changes keys after it has signed a Scheduled Transaction. 
+  - Scheduled Transactions will need to be re-signed with the new keys for them to succeed in this case.
+- Account deletion.
```
Comment (tinker-michaelj) [inline HIP/hip-0000-long-term-scheduled-transactions.md:123]: Could be generalized to "Invalidation of pending scheduled transaction"...even if a transaction was valid at `ScheduleCreate`, any number of things could happen (e.g., an account is deleted) that will cause it to fail at execution time.

**Review by tinker-michaelj — APPROVED** [review]
LGTM, a model HIP! ⭐

File: HIP/hip-0000-long-term-scheduled-transactions.md (hunk 3 above)
Comment (statop) [reply HIP/hip-0000-long-term-scheduled-transactions.md:125]: reworded :-)

**Review by tinker-michaelj — DISMISSED** [review]
LGTM!

**Review by sergmetelin — COMMENTED** [review]
LGTM

**Review by sergmetelin — APPROVED** [review]
LGTM

**Pull request merged**